src/engine/strat_engine/pool.rs

```diff
@@ -105,22 +109,46 @@ impl StratPool {
             redundancy: redundancy,
             thin_pool: thinpool_dev,
             mdv: mdv,
+            thindev_ids: ThinDevIdPool::new_from_ids(vec![]),
         };
 
         try!(pool.write_metadata());
 
         Ok(pool)
     }
 
+    /// Setup a StratPool using its UUID and the list of devnodes it has.
+    pub fn setup(uuid: PoolUuid, devnodes: &[PathBuf]) -> EngineResult<StratPool> {
+        let metadata = try!(try!(get_metadata(uuid, devnodes))
+                                .ok_or(EngineError::Engine(ErrorEnum::NotFound,
+                                                           format!("no metadata for pool {}",
+                                                                   uuid))));
+        let blockdevs = try!(get_blockdevs(&metadata, devnodes));
+        let (thinpool, mdv) = try!(get_dmdevs(uuid, &blockdevs, &metadata));
+        let filesystems = try!(get_filesystems(uuid, &thinpool, &mdv));
+        let thindev_ids =
+            ThinDevIdPool::new_from_ids(filesystems.iter().map(|x| x.thin_id()).collect());
+        Ok(StratPool {
+               name: metadata.name,
+               pool_uuid: uuid,
+               block_devs: BlockDevMgr::new(blockdevs),
+               filesystems: Table::new(),
```

Comment:
Whoops! I forgot, need to get the filesystems into the table. Otherwise, shoudl be good.